ggml/src/ggml-vulkan.cpp

```diff
@@ -1743,16 +1743,21 @@ static vk_device ggml_vk_get_device(size_t idx) {
         vk11_features.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_1_FEATURES;
         device_features2.pNext = &vk11_features;
 
+        VkPhysicalDevice16BitStorageFeatures storage_16bit;
+        storage_16bit.pNext = nullptr;
+        storage_16bit.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_16BIT_STORAGE_FEATURES;
+        vk11_features.pNext = &storage_16bit;
+
         VkPhysicalDeviceVulkan12Features vk12_features;
         vk12_features.pNext = nullptr;
         vk12_features.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_VULKAN_1_2_FEATURES;
-        vk11_features.pNext = &vk12_features;
+        storage_16bit.pNext = &vk12_features;
 
         vkGetPhysicalDeviceFeatures2(device->physical_device, &device_features2);
 
         device->fp16 = device->fp16 && vk12_features.shaderFloat16;
 
-        if (!vk11_features.storageBuffer16BitAccess) {
+        if (!(vk11_features.storageBuffer16BitAccess && storage_16bit.storageBuffer16BitAccess)) {
```

Comment:
```suggestion
        if (!vk11_features.storageBuffer16BitAccess || !storage_16bit.storageBuffer16BitAccess) {
```
Just a nitpick, but I think it's more readable this way.